### PR TITLE
admin-mode cookie issue fixed

### PR DESCRIPTION
**Pull request by yhtyyar94**
i did not remove all isAdminMode props just made it secure

### CODE REVIEW

**Comment by yhtyyar94** [comment]
I do not have access to the deployment logs, what is the issue?

**Comment by 7alip** [comment]
> @7alip I do not have access to the deployment logs, what is the issue?

@yhtyyar94 They are lint issues (Unsued vars, import orders etc). You might be able to see when you run `yarn lint`.

File: apps/samenvvv/pages/api/kv/hashtag-sentences.ts [hunk 1]
```diff
@@ -7,8 +9,20 @@ export const config = {
...
-
   try {
+    const { user, roles }: { user: User; roles: RoleType } = await unsealData(
+      req.cookies.get('iron-session')?.value as string,
+      {
+        password:
+          process.env['SECRET_COOKIE_PASSWORD'] ||
+          '12345678901234567890123456789012',
+      },
+    )
```
Comment (7alip) [inline apps/samenvvv/pages/api/kv/hashtag-sentences.ts:21]: Didn't `withIronSessionApiRoute` work with `edge` type api route?. We use it in other api routes but I'm not sure if it's working with `edge` routes. If it's not working, you can  pass password from `sessionOptions.password`

File: apps/samenvvv/pages/login.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { AdminLoginForm, LoginForm } from '@wsvvrijheid/ui'
+
+type PageProps = InferGetStaticPropsType<typeof getStaticProps>
+
+const LoginPage: FC<PageProps> = ({ seo }) => {
+  return (
+    <>
+      <NextSeo {...seo} />
+      <Box minH="inherit" h="full">
+        <LoginForm isSamenvvv={true} />
```
Comment (7alip) [inline apps/samenvvv/pages/login.tsx:18]: Prop name could be general, `isLoginOnly` or `hideLink`, `showLink` etc.

File: apps/samenvvv/pages/hashtags/[slug]/index.tsx [hunk 3]
```diff
@@ -121,7 +121,11 @@ export const getServerSideProps = async (
   const locale = context.locale as StrapiLocale
   const slug = context.params?.slug as string
   const { req, res, query } = context
-  const adminMode = getCookie('admin-mode', { req, res })
+  const adminMode: Auth | null = context.req.cookies['iron-session']
+    ? await unsealData(context.req.cookies['iron-session'] as string, {
+        password: process.env['SECRET_COOKIE_PASSWORD'] as string,
+      })
+    : null
```
Comment (7alip) [inline apps/samenvvv/pages/hashtags/[slug]/index.tsx:128]: I prefer removing isAdminMode here and check it in component level with `authContext`.

We have `checkAuth` function which fills authContext state. We call this function in AdminLayout and foundation Layout.

```tsx
const { checkAuth } = useAuthContext()

  useEffect(() => {
    checkAuth()
  }, [])
  ```
  
  You can add this to samen Layout as well. Then you will be able to get roles inside the component (PostSentencesModal or PostMakerTweetShare).
  
  ```tsx
  const { user, roles } = useAuthContext()
  ```
  
  You can also add `checkAuth` to login onSuccess just like we have in admin login.

**Review by 7alip — COMMENTED** [review]
Thanks

File: apps/samenvvv/pages/hashtags/[slug]/index.tsx [hunk 4]
```diff
@@ -102,8 +102,8 @@ const HashtagPage: FC<HashtagProps> = ({
       )}
       <Layout seo={seo}>
         <Container py={4} pos="relative">
-          {hasStarted || isAdminMode ? (
-            <PostMaker isAdminMode={isAdminMode} isIosSafari={isIosSafari} />
+          {hasStarted || roles.includes('admin') ? (
```
Comment (7alip) [inline apps/samenvvv/pages/hashtags/[slug]/index.tsx:105]: @yhtyyar94 I merged dev into your branch and also add admin role here.

**Review by 7alip — APPROVED** [review]
Thank you, you can merge

**Pull request merged**